http-api/src/lib.rs

```diff
@@ -740,6 +826,30 @@ fn remote_branch(branch_name: &str, peer_id: &PeerId) -> git::Branch {
     )
 }
 
+fn commit_hash_to_fingerprint(ctx: Context, sha1: &str) -> Option<String> {
+    let output = Command::new("git")
+        .current_dir(ctx.paths.git_dir()) // We need to place the command execution in the git dir
+        .args(["show", sha1, "--pretty='%GF'", "--raw"])
+        .output()
+        .expect("failed to execute git show command");
+
+    let mut result = output.stdout;
+
+    // If there is no stdout or if there is some stderr we return None early.
+    if result.is_empty() || !output.stderr.is_empty() {
+        return None;
+    }
+
+    // We match stdout against 'SHA256: which is what git show starts with if there is a ssh fingerprint.
+    match result.starts_with(b"'SHA256:") {
+        true => {
+            let fingerprint: Vec<u8> = result.splice(1..51, Vec::new()).collect();
```

Comment:
This seems a bit brittle, can we instead get the first line out of the result? I'd also skip the check for SHA256, it will simply not be found if it's not in the map.

Reply:
You should be able to do something like `io::BufReader::new(result).lines().next()` to get the first line.